autoupgrade.php

```diff
@@ -26,151 +26,161 @@
+
+        /* If a previous version of ajax-upgradetab.php exists, delete it */
+        if (file_exists($autoupgrade_dir.DIRECTORY_SEPARATOR.'ajax-upgradetab.php')) {
+            @unlink($autoupgrade_dir.DIRECTORY_SEPARATOR.'ajax-upgradetab.php');
+        }
+
+        /* Then, try to copy the newest version from the module's directory */
+        if (!@copy(dirname(__FILE__).DIRECTORY_SEPARATOR.'ajax-upgradetab.php', $autoupgrade_dir.DIRECTORY_SEPARATOR.'ajax-upgradetab.php')) {
+            return $this->_abortInstall(sprintf($this->l('Unable to copy ajax-upgradetab.php in %s'), $autoupgrade_dir));
+        }
+
+        /* Make sure that the XML config directory exists */
+        if (!file_exists(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml') &&
+        !@mkdir(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml', 0775)) {
+            return $this->_abortInstall(sprintf($this->l('Unable to create the directory "%s"'), _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml'));
+        } else {
+            @chmod(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml', 0775);
+        }
+
+        /* Create a dummy index.php file in the XML config directory to avoid directory listing */
+        if (!file_exists(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml'.DIRECTORY_SEPARATOR.'index.php') &&
+        (file_exists(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'index.php') &&
+        !@copy(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'index.php', _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml'.DIRECTORY_SEPARATOR.'index.php'))) {
+            return $this->_abortInstall(sprintf($this->l('Unable to create the directory "%s"'), _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'xml'));
+        }
+
+        return parent::install();
+    }
+
+    public function uninstall()
+    {
+        /* Delete the 1-click upgrade Back-office tab */
+        if ($id_tab = Tab::getIdFromClassName('AdminSelfUpgrade')) {
+            $tab = new Tab((int)$id_tab);
+            $tab->delete();
+        }
+
+        /* Remove the 1-click upgrade working directory */
+        self::_removeDirectory(_PS_ADMIN_DIR_.DIRECTORY_SEPARATOR.'autoupgrade');
+
+        return parent::uninstall();
+    }
+
+    public function getContent()
+    {
+        global $cookie;
+        header('Location: index.php?tab=AdminSelfUpgrade&token='.md5(pSQL(_COOKIE_KEY_.'AdminSelfUpgrade'.(int)Tab::getIdFromClassName('AdminSelfUpgrade').(int)$cookie->id_employee)));
+        exit;
+    }
+
+    /**
+    * Set installation errors and return false
+    *
+    * @param string $error Installation abortion reason
+    * @return boolean Always false
+    */
+    protected function _abortInstall($error)
+    {
+        $this->_errors[] = $error;
+
+        return false;
+    }
+
+    private static function _removeDirectory($dir)
```

Comment:
We prefer to use legacy methods to avoid to re-test the all module as it should be reworked "soon"